src/lender/adapters/mkr/clerk.sol

```diff
@@ -152,13 +162,15 @@ contract Clerk is Auth, Math {
         // protection value for the creditline increase coming from the junior tranche => amount by that the juniorAssetValue should be decreased
         uint protectionDAI = safeSub(overcollAmountDAI, amountDAI);
         // check if the new creditline would break the pool constraints
-        validate(0, protectionDAI, overcollAmountDAI, 0);
+        require((validate(0, protectionDAI, overcollAmountDAI, 0) == 0), "supply not possible, pool constraints violated");
         // increase MKR crediline by amount
         creditline = safeAdd(creditline, amountDAI);
     }
 
     // mint DROP, join DROP into cdp, draw DAI and send to reserve
     function draw(uint amountDAI) public auth active {
+        // make sure ther eis no collateral deficit before drawing out new DAI
+        require(collatDeficit() == 0, "please heal cdp first");
```

Comment:
I think we need to brainstorm about this require a bit.

Currently the coordinator considers the total creditline as available and would allow redeem on it.

In the executeEpoch transaction the reserve would fail to draw the DAI